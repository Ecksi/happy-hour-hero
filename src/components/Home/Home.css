* {
  font-family: 'Nunito', sans-serif; }

.homeContainer {
  width: 600px;
  margin: 50px auto 0 auto;
  text-align: center; }

.homeLogo {
  height: 250px; }

.homeContainer h2 {
  font-weight: 300;
  color: #555;
  font-size: 1.3rem;
  margin-top: 60px; }

.homeSearchForm {
  display: flex;
  flex-direction: column;
  position: relative; }

.homeSearchInput {
  margin-bottom: 20px;
  border-radius: 5px;
  height: 50px;
  border: 5px solid rgba(0, 0, 0, 0.1);
  font-size: 1rem;
  padding: 0 40px; }

.homeSearchInput::placeholder {
  color: #999;
  text-align: center; }

.homeSearchSubmit {
  height: 60px;
  background: #fdb82a;
  color: #fff;
  font-size: 1.2rem;
  border: none;
  border-radius: 5px; }

.homeSearchSubmit:hover {
  cursor: pointer; }

.fa-map-marker-alt {
  font-weight: 900;
  font-size: 1.7rem;
  color: #ccc;
  position: absolute;
  left: 30px;
  top: 15px; }

.fa-search {
  font-weight: 900;
  font-size: 1.7rem;
  color: #ccc;
  position: absolute;
  right: 30px;
  top: 15px; }

.fa-map-marker-alt:hover {
  cursor: pointer; }

.findLocationDropdown {
  position: relative;
  margin-top: -20px;
  margin-bottom: 30px;
  box-shadow: 1px 1px 1px 1px #eee;
  padding: 20px 80px;
  align-items: center;
  width: 73%;
  border-radius: 3px; }

.findLocationDropdown a {
  color: #1493ff;
  float: left; }

.findLocationDropdown a:hover {
  cursor: pointer; }

@media only screen and (max-width: 700px) {
  .findLocationDropdown {
    width: 240px; }
  .homeContainer {
    width: 400px; } }
