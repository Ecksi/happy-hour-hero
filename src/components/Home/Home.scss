* {
  font-family: 'Nunito', sans-serif;
}

.homeContainer {
  width: 600px;
  margin: 50px auto 0 auto;
  text-align: center;
}

.homeLogo {
  height: 250px;
}

.homeContainer h2 {
  font-weight: 300;
  color: #555;
  font-size: 1.3rem;
  margin-top: 60px;
}

.homeSearchForm {
  display: flex;
  flex-direction: column;
  position: relative;
}

.homeSearchInput {
  margin-bottom: 20px;
  border-radius: 5px;
  height: 50px;
  border: 5px solid rgba(0,0,0,0.1);
  font-size: 1rem;
  padding: 0 40px;
}

.homeSearchInput::placeholder {
  color: #999;
  text-align: center;
}

.homeSearchSubmit {
  height: 60px;
  background: #fdb82a;
  color: #fff;
  font-size: 1.2rem;
  border: none;
  border-radius: 5px;
}

.homeSearchSubmit:hover {
  cursor: pointer;
}

.fa-map-marker-alt {
  font-weight: 900;
  font-size: 1.7rem;
  color: #ccc;
  position: absolute;
  left: 30px;
  top: 15px;
}

.fa-search {
  font-weight: 900;
  font-size: 1.7rem;
  color: #ccc;
  position: absolute;
  right: 30px;
  top: 10px;
}

.fa-map-marker-alt:hover {
  cursor: pointer;
}

.findLocationDropdown {
  position: relative;
  margin-top: -20px;
  border-left: 1px solid #ccc;
  border-right: 1px solid #ccc;
  border-bottom: 1px solid #ccc;
  margin-bottom: 30px;
  padding: 20px 80px;
  -ms-flex-align: center;
  align-items: center;
}

.findLocationDropdown a {
  color: #1493ff;
  float: left;
}

.findLocationDropdown a:hover {
  cursor: pointer;
}