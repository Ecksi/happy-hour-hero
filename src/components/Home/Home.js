import React, { Component } from 'react';
import { connect } from 'react-redux';
import './Home.css';
import homeLogo from './images/home-logo.png';
import LocationAutocomplete from 'location-autocomplete';
import { googleApiKey } from '../../apiCalls/apiKeys/googleApiKey';
import { storeLocation } from '../../actions';
import GeoLocator from '../GeoLocator/GeoLocator';

class Home extends Component {
  constructor (props) {
    super(props);

    this.state = {
      city: '',
      state: '',
      zip: '',
      findLocationDropdown: false
    };
  }

  onChangeHandler = (event) => {
    const { name, value } = event.target;
    
    this.setState({
      [name]: value
    });
  }

  onDropdownSelect = async (component) => {
    const { zip } = this.state;

    if (isNaN(zip)) {
      const place = component.autocomplete.getPlace();
      const city = place.vicinity;
      const state = place.address_components[2].short_name;
     
      this.setState({
        city,
        state
      });
    }
  }

  handleSubmit = async (event) => {
    let latitude;
    let longitude;
    let location;

    event.preventDefault();
    const { city, state, zip } = this.state;

    if (city && state) {
      location = `${city}+${state}`;
    } else if (!isNaN(parseInt(zip)) && zip.length === 5) {
      location = zip;
    }

    const response = await fetch(`https://maps.googleapis.com/maps/api/geocode/json?address=${location}&key=${googleApiKey}`);
    const data = await response.json();
    latitude = data.results[0].geometry.location.lat;
    longitude = data.results[0].geometry.location.lng;

    this.props.storeLocation(city, state, zip, null, longitude, latitude);

    this.props.history.push('/HappyHours');
  }

  getInnerRef = (ref) => {
    this.innerRef = ref;
  }

  getLocation = () => {
    this.innerRef && this.innerRef.getLocation();
    
    setTimeout(async () => {
      const { longitude, latitude } = this.innerRef.state.coords;
   
      const response = await fetch(`https://maps.googleapis.com/maps/api/geocode/json?latlng=${latitude},${longitude}&key=${googleApiKey}`);
      const data = await response.json();
      const address = data.results[0].formatted_address;

      this.props.storeLocation(null, null, null, address, longitude, latitude);
      this.props.history.push('/HappyHours');
    }, 10000);
  }

  handleSearchInput = () => {
    if (this.state.zip === "") {
      this.setState({
        findLocationDropdown: true
      });
    } else {
      this.setState({
        findLocationDropdown: false
      });  
    }
  }

  render() {
    const { getInnerRef, getLocation } = this;

    return (
      <section className="homeContainer">
        <img src={homeLogo} className="homeLogo" alt="Happy Hour Hero Logo" />
        <h2>Find your happy hour:</h2>
        <form className="homeSearchForm" onSubmit={this.handleSubmit}>
          <i className="fas fa-search"></i>
          <LocationAutocomplete
            name="zip"
            className="homeSearchInput" 
            placeholder="Enter a restaurant or location" 
            targetArea="City, State"
            locationType="(cities)" 
            googleAPIKey={googleApiKey}
            onChange={this.onChangeHandler}
            onKeyDown={this.handleSearchInput}
            onFocus={this.handleSearchInput}
            onDropdownSelect={this.onDropdownSelect}
          />
          <div className="findLocationDropdown" style={{display: this.state.findLocationDropdown ? 'block' : 'none' }}>
            <GeoLocator ref={getInnerRef} />
            <i className="fas fa-map-marker-alt" onClick={this.getLocation} ></i>
            <a onClick={this.getLocation}>Current Location</a>
          </div>
          <input className="homeSearchSubmit" type="submit" value="Submit" />
        </form>
      </section>
    );
  }
}

export const mapDispatchToProps = (dispatch) => ({
  storeLocation: (city, state, zip, address, longitude, latitude) => {
    return dispatch(storeLocation(city, state, zip, address, longitude, latitude));
  }
});

export default connect(null, mapDispatchToProps)(Home);
