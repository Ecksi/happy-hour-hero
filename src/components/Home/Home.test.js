import React from 'react';
import { shallow } from 'enzyme';
import Home from './Home';

describe('Home', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Home />);
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});