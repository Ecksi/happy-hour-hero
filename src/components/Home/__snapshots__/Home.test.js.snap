// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="Happy Hour Hero Logo"
          className="homeLogo"
          src="home-logo.png"
/>,
        <h2>
          Find your happy hour:
</h2>,
        <withRouter(Connect(SearchBar)) />,
      ],
      "className": "homeContainer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Happy Hour Hero Logo",
          "className": "homeLogo",
          "src": "home-logo.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Find your happy hour:",
        },
        "ref": null,
        "rendered": "Find your happy hour:",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="Happy Hour Hero Logo"
            className="homeLogo"
            src="home-logo.png"
/>,
          <h2>
            Find your happy hour:
</h2>,
          <withRouter(Connect(SearchBar)) />,
        ],
        "className": "homeContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Happy Hour Hero Logo",
            "className": "homeLogo",
            "src": "home-logo.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Find your happy hour:",
          },
          "ref": null,
          "rendered": "Find your happy hour:",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
